credentials/credentials_test.go

```diff
@@ -44,12 +46,53 @@ func (mockSecretManagerErr) GetSecretValueWithContext(context.Context, *secretsm
 }
 
 func TestIsSecretConfigured(t *testing.T) {
-	OverrideSecretsManager(mockSecretManager{})
 	ctx := context.Background()
-	assert.True(t, IsSecretConfigured(ctx, &config.Configuration{}))
+	assert.False(t, IsSecretConfigured(ctx, &config.Configuration{}))
+
+	OverrideSecretsManager(mockSecretManager{})
+	assert.True(t, IsSecretConfigured(ctx, &config.Configuration{
+		LicenseKeySecretId: "testSecretName",
+	}))
 
 	OverrideSecretsManager(mockSecretManagerErr{})
-	assert.False(t, IsSecretConfigured(ctx, &config.Configuration{}))
+	assert.False(t, IsSecretConfigured(ctx, &config.Configuration{
+		LicenseKeySecretId: "testSecretName",
+	}))
+}
+
+type mockSSM struct {
+	ssmiface.SSMAPI
+}
+
+func (mockSSM) GetParameterWithContext(context.Context, *ssm.GetParameterInput, ...request.Option) (*ssm.GetParameterOutput, error) {
+	return &ssm.GetParameterOutput{
+		Parameter: &ssm.Parameter{
+			Value: aws.String("foo"),
+		},
+	}, nil
+}
+
+type mockSSMErr struct {
+	ssmiface.SSMAPI
+}
+
+func (mockSSMErr) GetParameterWithContext(context.Context, *ssm.GetParameterInput, ...request.Option) (*ssm.GetParameterOutput, error) {
+	return nil, fmt.Errorf("Something went wrong")
+}
+
+func TestIsParameterConfigured(t *testing.T) {
+	ctx := context.Background()
+	assert.False(t, IsParameterConfigured(ctx, &config.Configuration{}))
+
+	OverrideSSM(mockSSM{})
+	assert.True(t, IsParameterConfigured(ctx, &config.Configuration{
+		LicenseKeyParameterName: "testParameterName",
+	}))
+
+	OverrideSSM(mockSSMErr{})
+	assert.False(t, IsParameterConfigured(ctx, &config.Configuration{
+		LicenseKeyParameterName: "testParameterName",
+	}))
 }
 
 func TestGetNewRelicLicenseKey(t *testing.T) {
```

Comment:
I feel like it'd be useful to add tests here to ensure all the fallbacks work as we expect.